cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkRuntimeContext.java

```diff
@@ -226,7 +226,7 @@ DynamicDatasetCache getDatasetCache() {
    * Returns the {@link WorkflowProgramInfo} if the spark program is running inside a workflow.
    */
   @Nullable
-  WorkflowProgramInfo getWorkflowProgramInfo() {
+  WorkflowProgramInfo getWorkflowInfo() {
```

Comment:
For MapReduce, this method is now defined in the MapReduceContext interface, and implemented by BasicMapReduceContext and others. What is the Spark equivalent of the MapReduceContext, the interface that is exposed to actual Spark program and its lifecycle methods? It should be defined in that interface and then implemented with @Override here